drizzlepac/haputils/catalog_utils.py

```diff
@@ -1180,9 +1183,12 @@ def identify_sources(self, **pars):
                         reg_sources.remove_column("daofind_mag")
 
                     if sources is None:
+                        last_used_id = reg_sources['id'][-1]
                         sources = reg_sources
                     else:
+                        reg_sources['id'][:] +=  last_used_id
```

Comment:
In the future, I think you can use reg_sources['id'] instead of reg_sources['id'][:], but everything else looks fines to me. I was able to confirm that the code works using program "ictj64".